modules/library/src/main/scala/zio/elasticsearch/ElasticAggregationResponse.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.json._
+import zio.json.ast.Json.Obj
+import zio.json.ast._
+import zio.json.internal.RetractReader
+
+sealed trait ElasticAggregationResponse
+
+final case class TermsAggregationResponse(
+  @jsonField("doc_count_error_upper_bound")
+  docErrorCount: Int,
+  @jsonField("sum_other_doc_count")
+  sumOtherDocCount: Int,
+  buckets: List[TermsAggregationBucket]
+) extends ElasticAggregationResponse
+
+object TermsAggregationResponse {
+  implicit val decoder: JsonDecoder[TermsAggregationResponse] = DeriveJsonDecoder.gen[TermsAggregationResponse]
+}
+
+sealed trait ElasticAggregationBucket
+
+final case class TermsAggregationBucket(
+  key: String,
+  @jsonField("doc_count")
+  docCount: Int,
+  subAggregations: Option[Map[String, ElasticAggregationResponse]] = None
+) extends ElasticAggregationBucket
+
+object TermsAggregationBucket {
+  implicit val decoder: JsonDecoder[TermsAggregationBucket] = (trace: List[JsonError], in: RetractReader) => {
+    val json = JsonDecoder[Json].unsafeDecode(trace, in)
```

Comment:
Instead of doing it like this, maybe consider:
```scala
  implicit val decoder: JsonDecoder[TermsAggregationBucket] = JsonDecoder[Obj].mapOrFail(obj => ...)
```
I think it would make sense if the semantics were the same, which should be looked into.

Reply:
Also, I've seen you use `Obj.decoder` in a few places, so either use that in this case too for consistency, or change the rest to `JsonDecoder[Obj]`.